liquidity/liquidity.go

```diff
@@ -457,6 +505,142 @@ func (m *Manager) ForceAutoLoop(ctx context.Context) error {
 	}
 }
 
+// dispatchBestEasyAutoloopSwap tries to dispatch a swap to bring the hot wallet
+// size back to its target.
+func (m *Manager) dispatchBestEasyAutoloopSwap(ctx context.Context) error {
+	// If autoloop is turned off, return early.
+	if m.params.Autoloop == false {
+		return nil
+	}
+
+	// Retrieve existing swaps.
+	loopOut, err := m.cfg.ListLoopOut()
+	if err != nil {
+		return err
+	}
+
+	loopIn, err := m.cfg.ListLoopIn()
+	if err != nil {
+		return err
+	}
+
+	// Get a summary of our existing swaps so that we can check our autoloop
+	// budget.
+	summary, err := m.checkExistingAutoLoops(ctx, loopOut, loopIn)
+	if err != nil {
+		return err
+	}
+
+	err = m.checkSummaryBudget(summary)
+	if err != nil {
+		return err
+	}
+
+	_, err = m.checkSummaryInflight(summary)
+	if err != nil {
+		return err
+	}
+
+	// Get all channels in order to calculate current hot wallet balance.
+	channels, err := m.cfg.Lnd.Client.ListChannels(ctx, false, false)
+	if err != nil {
+		return err
+	}
+
+	hotWalletBalance := btcutil.Amount(0)
+	for _, channel := range channels {
+		hotWalletBalance += channel.LocalBalance
+	}
+
+	// Since we're only autolooping-out we need to check if we are below
+	// the target, meaning that we already meet the requirements.
+	if hotWalletBalance <= m.params.EasyAutoloopTarget {
+		log.Debugf("hot wallet balance %v below target %v",
+			hotWalletBalance, m.params.EasyAutoloopTarget)
+		return nil
+	}
+
+	restrictions, err := m.cfg.Restrictions(ctx, swap.TypeOut)
+	if err != nil {
+		return err
+	}
+
+	// Calculate the amount that we want to loop out. Check against the max
+	// swap amount of terms.
+	amount := hotWalletBalance - m.params.EasyAutoloopTarget
+	if amount > restrictions.Maximum {
```

Comment:
We're attempting swaps from biggest to smallest, so got to check that our highest values respect the max limit, because those are most likely to be picked next.

If we pick a channel for a swap and the amount is <= 250k that means it also was our best choice so lets let the server decide